doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1265 @@
+```
+ALARM_STATS Table:
+==============================
+
+Key                       : id
+
+id                        : key {state}
+alarms                    : Number of active alarms {uint64}
+critical                  : Number of alarms of severity 'critical' {uint64}
+major                     : Number of alarms of severity 'major' {uint64}
+minor                     : Number of alarms of severity 'minor' {uint64}
+warning                   : Number of alarms of severity 'warning' {uint64}
+informational             : Number of alarms of severity 'informational' {uint64}
+
+127.0.0.1:6379[6]> hgetall "ALARM_STATS|state"
+ 1) "alarms"
+ 2) "1"
+ 3) "critical"
+ 4) "1"
+ 5) "major"
+ 6) "0"
+ 7) "minor"
+ 8) "0"
+ 9) "warning"
+10) "0"
+
+```
+### 3.1.8 Pull Model
+All NBIs - CLI, REST and gNMI - can pull contents of current alarm table and event history table.
+The following filters are supported:
+- ALL ( pulls all alarms)
+- Severity.
+- Recent alarms (eg., last 5 minutes, one hour, one day).
+- Records between two timestamps, one timestamp and end, and   beginning and a timestamp.
+- All records between two Sequence Numbers (incl end and begin)
+
+### 3.1.9 Supporting third party containers
+To support third party components ( e.g. FRR, teamd, DHCP Relay, LLDPd, ntpd etc ) which can not be modified to raise events, the following options are considered 
+and are being evaluated.
+1.  Patch the components
+    Create a patch for these components by adding libeventnotify library and invoke the API. This however, requires these patches need to be maintained in the code forever.
+
+2.  Listen to syslog messages
+    As many of these components raises syslog messages on an important event, a listener can be implemented to read incoming syslog messages and raise 
+    events based on the message.
+    This however is heavy on performance due to the fact that listener has to parse each syslog message. Also listener need to maintain a map of messages to 
+    event-id and need to be aware of resource and other specific details. It need to be aware of nuances of alarm raising/clearing if the component follows
+    any specific logic. 
+
+Approach 1 is preferred.
+
+## 3.2 DB Changes
+### 3.2.1 EVENT DB
+A new instance, redis4, is created and EVENT DB uses the new instance.
+The following tables uses Event DB.
+Table EVENTPUBSUB is used for applications to write events and for eventd to access and process them.
+Event History Table (EVENT) and Current Alarm Table (ALARM) are used to house events and alarms respectively.
+To maintain various statistics of events, these two tables are used : EVENT_STATS and ALARM_STATS.
+
+EVPROFILE table is used by mgmt-framework to communicate name of the custom event profile when configured through NBI. 
+Eventd reads the file name from this table and merges it with its static_event_map.
+
+## 3.3 User Interface
+### 3.3.1 Data Models
+
+The following is SONiC yang for events.
+```
+module: sonic-event
```

Comment:
For Yang model i think that sub group should lead and approve this HLD.
@zhangyanzhao please assign it to someone from the subgroup.